codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -252,7 +252,7 @@ public int onDataRead(final ChannelHandlerContext ctx, int streamId, ByteBuf dat
             Http2LocalFlowController flowController = flowController();
             try {
                 // If we should apply flow control, do so now.
-                if (shouldApplyFlowControl) {
+                if (shouldApplyFlowControl && bytesToReturn > 0) {
```

Comment:
> sorry, I read your original comment via e-mail so I didn't realize you had already provided the link ;)

No problem.  Just to confuse things even more...I sometimes edit my comments to clarify/correct after the emails go out anyhow :). I always double check comments I get through email on github.